chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -158,20 +217,39 @@ def self.build_node_map(node)
             count_map[speed] = count + 1
           end
         end
-
+        # if_remap now contains something like
+        #   {
+        #     "100m1" => "em1",
+        #     "1g1"   => "em1",
+        #     "10g1"  => "em1",
+        #     "100m2" => "em2",
+        #     "1g2"   => "em2",
+        #     "10g2"  => "em2",
+        #     "10m1"  => "em3",
+        #     "100m3" => "em3",
+        #     "1g3"   => "em3",
+        #     "10m2"  => "em4",
+        #     "100m4" => "em4",
+        #     "1g4"   => "em4"
+        #   }
         ans = {}
-        conduits.each do |k,v|
+        conduits.each do |conduit, conduit_data|
           hash = {}
-          v.each do |mk, mv|
-            if mk == "if_list"
-              hash["if_list"] = v["if_list"].map do |if_ref|
-                map_if_ref(if_remap, if_ref)
+          conduit_data.each do |conduit_key, conduit_val|
+            if conduit_key == "if_list"
+              hash["if_list"] = conduit_data["if_list"].map do |if_ref|
+                r = map_if_ref(if_remap, if_ref)
+                if r.nil?
+                  Chef::Log.warn "Unable to find interface for #{node.fqdn} " \
+                                 "matching #{if_ref} out of #{if_remap}"
```

Comment:
> I such a case this will cause a lot of unnecessary warnings (build_node_map is called quite frequently) in the logs. I fear that it will just confuse people.

Yeah OK, fair point.

> Wouldn't be better to throw a more useful error message for cases where the conduit is actually used on a node

Sure, that's why this PR also adds [this error](https://github.com/aspiers/crowbar-core/blob/12da87dd0bbca677d84739ec831fb31dae5900cf/chef/cookbooks/provisioner/recipes/update_nodes.rb#L24).

> Or even when trying to allocate/enable a network already (i.e. in the rails app)

You mean, somewhere like in `NetworkService#allocate_ip_by_type`?  That seems to get all its network info from the databag.